crates/node-api/src/engine/mod.rs

```diff
@@ -196,6 +216,40 @@ pub fn validate_payload_timestamp(
     Ok(())
 }
 
+#[cfg(feature = "optimism")]
+/// Validates the presence of the `withdrawals` field according to the payload timestamp.
+/// After Canyon, withdrawals field must be [Some].
+/// Before Canyon, withdrawals field must be [None];
+pub fn optimism_validate_withdrawals_presence(
+    chain_spec: &ChainSpec,
+    version: EngineApiMessageVersion,
+    timestamp: u64,
+    has_withdrawals: bool,
+) -> Result<(), AttributesValidationError> {
+    let is_shanghai = chain_spec.fork(Hardfork::Canyon).active_at_timestamp(timestamp);
```

Comment:
this is basically the same function as eth.

I think it's fine to duplicate this, though we should reference here that Canyon activates the Shanghai eips, see issue.

Reply:
added some docs and a link to the EIPs included in canyon